### PR TITLE
Fix ashift crashes while removing lines

### LINKED ISSUE #12529: dt crashes when right-clicking manually drawn perspective lines
**Describe the bug/issue**

I can consistently make darktable crash when using the "crop & rotate" module and the option "manually draw structure lines". The program quits without an error message and any changes made to the image are lost as a result.

I am using Windows 10 and the latest 4.0.1 version of darktable. FYI, in this older computer with an Intel i3-6100 (Skylake) and only the integrated graphics, I have to use dt with "--disable-opencl", otherwise it won't launch at all.

**To Reproduce**

1. Open a random image (mine was a random JPEG)
2. Click on the "rotate & perspective" module and then click on the pen to "manually draw structure lines"
3. Draw a line in the image
4. Right-click at least twice inside one of the 'circles' of that line
5. Draw another line

**Screencast**

https://user-images.githubusercontent.com/15838230/191051930-093f3ef8-785f-44d3-ab20-2486e2be5e62.mp4

* darktable version : 4.0.1
* OS                : Win10 (19044.2006)
* Memory            : 12 GB
* Graphics card     : Intel HD Graphics 530
* OpenCL activated  : no

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Found two bugs here
1. avoid out-of-array access while removing line segments
2. use dt_alloc_align_float and dt_free_align for image data and we make use of vector-optimized functions so we want to make sure

While being here
- added some safety checks while allocating
- avoid g_malloc0 and use calloc instead to safely use free()
- changed int to gboolean where appropriate

Here fixes #12529

### CODE REVIEW

**Review by TurboGit — APPROVED** [review]
Looks good, thanks!

**Review by TurboGit — CHANGES_REQUESTED** [review]
Makes dt crash, I open an image and clicked on the auto line detection of ashift.

**Comment by TurboGit** [comment]
Sadly the backtrace is not really helpful :

```
(gdb) bt
#0  dt_iop_image_copy._omp_fn.0 ()
    at /home/obry/dev/builds/c-darktable/x86_64-linux-gnu-default/src/src/common/imagebuf.c:171
#1  0x00007ffff770d36e in  () at /lib/x86_64-linux-gnu/libgomp.so.1
#2  0x00007ffff78a63ec in start_thread (arg=<optimized out>)
    at ./nptl/pthread_create.c:444
#3  0x00007ffff7926a4c in clone3 ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

File: src/iop/ashift.c [hunk 1]
```diff
@@ -1753,7 +1753,7 @@ static int _get_structure(dt_iop_module_t *module,
     scale = g->buf_scale;
 
     // create a temporary buffer to hold image data
-    buffer = malloc(sizeof(float) * 4 * (size_t)width * height);
+    buffer = dt_alloc_align_float((size_t)width * height);
```
Comment (TurboGit) [inline src/iop/ashift.c:1756]: missing 4 ?

Comment (jenshannoschwalm) [reply src/iop/ashift.c]: Thanks!

**Comment by jenshannoschwalm** [comment]
force-push squashed commit with the bug spotted by @TurboGit

**Review by TurboGit — APPROVED** [review]
Works for me now, thanks!

**Pull request merged**